include/boost/geometry/formulas/meridian_direct.hpp

```diff
@@ -84,7 +84,7 @@ class meridian_direct
             {
                 result.reverse_azimuth =  pi;
             }
-            else if (result.lat2 < -half_pi &&
+            else if ((result.lat2 < -half_pi) &&
```

Comment:
Weird that it warns for this, and not for the rest. We never have brackets here, unless and/or are combined.
But anyway - if it fixes a warning, it's fine.

Reply:
I couldn't really make sense of it and the warning is only shown with GCC IIRC. `lat2`, assuming I didn't miss an alternative definition of the result type is just not a templated field. My only guess (which I don't find convincing) is that it maybe misreads part of
```
else if (result.lat2 < -half_pi &&
         result.lat2 >  -one_and_a_half_pi)

```
As `result.lat2</*boolean template argument*/> - one_and_a_half_pi`. But then the logic that decides whether to emit a warning here would have to be different from the logic that correctly interprets and compiles that code and I would not expect a bug like that in GCC and I have no experience debugging GCC.

Reply:
Are you getting the same warning with gcc 12?

Reply:
Yes, I get the warning with both gcc12 and gcc13.

Reply:
Changing the second line of the condition to `-one_and_a_half_pi < result.lat2` also makes the warning disappear.

Reply:
Interesting...

Reply:
Could it be that it somehow interprets `< -half_pi && result.lat2 >` as a template?
Maybe you can try to reverse the conditions
```
(result.lat2 >  -one_and_a_half_pi 
 && result.lat2 < -half_pi)
         
```

Reply:
or probably not, it was a warning. The reversal you suggested is also fine.